src/jit/lsra.cpp

```diff
@@ -2683,6 +2683,9 @@ regMaskTP LinearScan::getKillSetForNode(GenTree* tree)
             {
                 assert(tree->AsObj()->gtGcPtrCount != 0);
                 killMask = compiler->compHelperCallKillSet(CORINFO_HELP_ASSIGN_BYREF);
+#ifdef _TARGET_ARM64_
+                killMask |= REG_WRITE_BARRIER_SRC_BYREF | REG_WRITE_BARRIER_DST_BYREF;
+#endif // _TARGET_ARM64_
```

Comment:
You are just adding R12 and R13 back to the kill set here.
 Better to just include them in the killset returned by compHelperCallKillSet instead.